src/main/scala/com/typesafe/sbteclipse/core/EclipsePlugin.scala

```diff
@@ -310,14 +297,66 @@ object EclipsePlugin {
     val All = Default
   }
 
-  object EclipseProjectFlavor extends Enumeration {
+  object EclipseProjectFlavor {
+    import scalaz.{ Success, Failure }
+    import scalaz.Scalaz.{ state => _, _ }
+    abstract class Value {
+      def resolved(ref: ProjectRef, state: State): Value = this
+      def scalacOptions(ref: ProjectRef, state: State): Validation[Seq[(String, String)]]
+      def defaultClasspathTransformerFactories: Seq[EclipseTransformerFactory[RewriteRule]]
+    }
+    case object Java extends Value {
+      def scalacOptions(ref: ProjectRef, state: State): Validation[Seq[(String, String)]] = Success(Seq.empty)
+      def defaultClasspathTransformerFactories: Seq[EclipseTransformerFactory[RewriteRule]] = Seq.empty
+    }
+    case object ScalaIDE extends Value {
+      def scalacOptions(ref: ProjectRef, state: State): Validation[Seq[(String, String)]] = {
+        // Here we have to look at scalacOptions *for compilation*, vs. the ones used for testing.
+        // We have to pick one set, and this should be the most complete set.
+        (Eclipse.evaluateTask(Keys.scalacOptions in sbt.Compile, ref, state) |@| Eclipse.settingValidation(Keys.scalaVersion in ref, state)) { (options, version) =>
+          val ideSettings = Eclipse.fromScalacToSDT(options)
+          util.ScalaVersion.parse(version).settingsFrom(ideSettings.toMap).toSeq
+        } map { options => if (options.nonEmpty) ("scala.compiler.useProjectSettings" -> "true") +: options else options }
+      }
+
+      def defaultClasspathTransformerFactories: Seq[EclipseTransformerFactory[RewriteRule]] =
+        Seq(EclipseRewriteRuleTransformerFactory.ClasspathDefault) // replace scala lib/compiler jars with scala container bundled in scala-ide
+    }
+
+    case object Autodetect extends Value {
+      override def resolved(ref: ProjectRef, state: State): Value = {
+        lazy val projectName = Eclipse.setting(Keys.name in ref, state)
+        state.log.debug(s"Project $projectName has $this Eclipse flavor.")
+        // Considering both the compile and test classpath, because if either of the two depends on the scala-library, then we need to return ScalaIDE as the resolved flavor.
+        val classpath = (Eclipse.evaluateTask(Keys.fullClasspath in sbt.Compile, ref, state) |@| Eclipse.evaluateTask(Keys.fullClasspath in sbt.Test, ref, state)) { (compileCp, testCp) =>
+          compileCp ++ testCp
+        }
+        classpath match {
+          case Failure(f) =>
+            state.log.debug(s"Failed to build classpath for $projectName. Error was: $f")
+            this
```

Comment:
We need to pick a default that is not `Autodetect` (the reason is that calling `defaultClasspathTransformerFactories` on Autodetect throws an exception). We should probably return the `ScalaIDE` flavour in this case.